### PR TITLE
:sparkles: gVCF support

### LINKED ISSUE #43: Add node for gVCF files
Initial name suggestion is `germline_variation_catalog`. 

It looks like the alignment workflow for the germline samples will output both the CRAM and gVCF, so it should link back to the `alignment_workflow` in a one to one fashion. And then I think it could potentially be used in a many to many fashion to generate joint called VCFs, so should be linked from `germline_mutation_calling_workflow`.

### PR DESCRIPTION
**Pull request by allisonheath**
~cc @philloooo for use case for oneOf required links~

can use the exclusive link subgroups, but that is an API level validation, still might be better to have some of that functionality built into the JSON schema

closes #43

### CODE REVIEW

File: gdcdictionary/schemas/germline_mutation_calling_workflow.yaml [hunk 1]
```diff
@@ -44,6 +51,10 @@ properties:
   workflow_type:
     term:
       $ref: "_terms.yaml#/workflow_type"
-    enum: [ "HaplotypeCaller" ]
+    enum:
+      - "HaplotypeCaller"
+      - "FreeBayes"
```
Comment (dmiller15) [inline gdcdictionary/schemas/germline_mutation_calling_workflow.yaml]: For consistency, the quotes aren't really necessary. GDCAPI will cast them as a string.

File: gdcdictionary/examples/valid/germline_mutation_calling_workflow.json [hunk 2]
```diff
@@ -4,6 +4,9 @@
     "workflow_link": "github.com/<owner>/<project>/commit/<hash>",
     "workflow_type": "HaplotypeCaller",
     "aligned_reads_files": {
-	"submitter_id": "GDC_generated_bam_#"
+	"submitter_id": "generated_cram_#"
+    },
+    "germline_variation_catalogs": {
+    "submitter_id": "generated_gVCF_#"
```
Comment (dmiller15) [inline gdcdictionary/examples/valid/germline_mutation_calling_workflow.json]: indent?

File: gdcdictionary/schemas/germline_mutation_calling_workflow.yaml [hunk 3]
```diff
@@ -27,12 +27,19 @@ links:
...
     multiplicity: many_to_many
     required: true
+  - name: germline_variation_catalogs
+    backref: germline_mutation_calling_workflows
+    label: performed_on
+    target_type: germline_variation_catalog
+    multiplicity: many_to_many
+    required: true
 
+#should be oneOf aligned_reads_files or germline_variation_catalogs, but not supported
```
Comment (dmiller15) [inline gdcdictionary/schemas/germline_mutation_calling_workflow.yaml]: Wait, why couldn't you do an exclusive subgroup? I understand the required props wouldn't work but I'm pretty sure the `uc-cdis/gdcapi` has the same functionality as `NCI-GDC`

**Review by dmiller15 — CHANGES_REQUESTED** [review]

File: gdcdictionary/schemas/germline_mutation_calling_workflow.yaml [hunk 4]
```diff
@@ -21,18 +21,26 @@ systemProperties:
...
+        label: performed_on
+        target_type: aligned_reads
+        multiplicity: many_to_many
+        required: true
+      - name: germline_variation_catalogs
+        backref: germline_mutation_calling_workflows
+        label: performed_on
+        target_type: germline_variation_catalog
+        multiplicity: many_to_many
+        required: true
```
Comment (dmiller15) [inline gdcdictionary/schemas/germline_mutation_calling_workflow.yaml]: Wait, is this allowed? Aren't you requiring both links exist? That would conflict with the `exclusive: true` wouldn't it? 😕

**Review by dmiller15 — CHANGES_REQUESTED** [review]

File: gdcdictionary/schemas/germline_mutation_calling_workflow.yaml (hunk 4 above)
Comment (allisonheath) [reply gdcdictionary/schemas/germline_mutation_calling_workflow.yaml]: Yup - snuck in there with some copy/paste I think - fixed

**Review by dmiller15 — APPROVED** [review]
👍

**Pull request merged**